### PR TITLE
[Merged by Bors] - intelliJ: use custom delve on macos

### LINKED ISSUE #695: use custom delve in IntelliJ
Following https://github.com/metalbear-co/mirrord/pull/694 we need to do the same for IntelliJ.
In case of macOS + Go, we edit the delve command line to use our delve instead of the one provided by homebrew/other means.

### PR DESCRIPTION
**Pull request by infiniteregrets**
closes #695

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -260,10 +260,20 @@ jobs:
         build_binaries_macos,
         release_docker_image,
       ]
-    runs-on: ubuntu-latest
+    runs-on: macos-12
```
Comment (eyalb181) [inline .github/workflows/release.yaml:263]: Does the entire intellij release step run on macos now? Was it tested?

Comment (aviramha) [reply .github/workflows/release.yaml:263]: wait.. no need to do it. run it on ubuntu, just wget the version like the other job does.

**Review by aviramha — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (infiniteregrets) [reply .github/workflows/release.yaml:263]: oh right, just saw the vscode fix was already done

**Review by aviramha — APPROVED** [review]
LGTM

**Pull request closed without merging**